File: scala/scala-start/src/main/scala/test/interface/Student.scala
package test.interface

/**
  * Created by saligia on 17-9-21.
  */
class Student (var name :String) extends Persion{

  override def getName(): String = this.name
  override def setName(name: String): Unit = {
    this.name = name
  }
  override def getMessage(): String = {
    return "name : " + this.name
  }

  def update(name:String,value:String):Unit = {
    println(name+ ":" + value)
  }

  def apply(name: String): Unit={
    println("name : " + name)
  }
}

File: cpp_primer_study/src/class/student.cpp
//
// Created by norma.eva on 2019-02-20.
//

#include "student.h"

extern u_int8_t test;

Student::Student(std::string name, u_int8_t age, std::string v_class)
    :Persion(name,age){
    this->v_class = v_class;
}

std::string Student::getClass() {
    return this->v_class;
}
